### PR TITLE
Update module

### PR DESCRIPTION
**Pull request by HesyRa**
Fix error when edit current user and optimize module

### CODE REVIEW

**Comment by HesyRa** [comment]
[Fix edit current user email error](https://trello.com/c/VGm9VDco/240-fix-edit-current-user-email-error)

**Review by Xapp73 — APPROVED** [review]

File: src/Service/AuthAdapter.php [hunk 1]
```diff
@@ -230,6 +229,31 @@ protected function validateUser(?UserInterface $user): Result
             null,
             ['Invalid credentials.']);
     }
-}
+    /**
+     * Find user by password reset token
+     * @param string $token
+     * @return mixed
+     */
+    public function getUserByToken(string $token) {
```
Comment (ikovalyov) [inline src/Service/AuthAdapter.php]: { from new line in functions definitions here and later

File: src/Service/AuthAdapter.php [hunk 2]
```diff
@@ -230,6 +229,31 @@ protected function validateUser(?UserInterface $user): Result
...
+        return $this->entityManager->getRepository(User::class)
+            ->findOneByEmail($email);
+    }
 
+    /**
+     * Get STATUS_RETIRED
+     * @return int
+     */
+    public function getRetiredStatus(){
+        return User::STATUS_RETIRED;
```
Comment (ikovalyov) [inline src/Service/AuthAdapter.php]: Why not use User::STATUS_RETIRED instead of this function?

File: src/Service/UserManager.php [hunk 3]
```diff
@@ -164,20 +164,21 @@ public function createAdminUserIfNotExists()
     public function hasRole($email, $roles)
     {
         /** @var User $user */
-        $user = $this->entityManager->getRepository(User::class)->findOneByEmail($email);
+        $user = $this->getUserByEmail($email);
         if(in_array($user->getRoleName(),$roles, true)) {
```
Comment (ikovalyov) [inline src/Service/UserManager.php]: return in_array($user->getRoleName(),$roles, true)

**Review by ikovalyov — CHANGES_REQUESTED** [review]

File: src/Service/AuthAdapter.php (hunk 2 above)
Comment (HesyRa) [reply src/Service/AuthAdapter.php]: Для переопределения функции наследником.

**Pull request merged**